qiita_db/metadata_template.py

```diff
@@ -620,23 +620,37 @@ def __setitem__(self, column, value):
         """
         conn_handler = SQLConnectionHandler()
 
-        exists = conn_handler.execute_fetchone("""
+        # try dynamic tables
+        exists_dynamic = conn_handler.execute_fetchone("""
             SELECT column_name
             FROM information_schema.columns
             WHERE table_name='{0}'
                 AND table_schema='qiita'
                 AND column_name='{1}'""".format(self._dynamic_table, column))
-
-        if exists is None:
+        # try required_sample_info
+        exists_required = conn_handler.execute_fetchone("""
+            SELECT column_name
+            FROM information_schema.columns
+            WHERE table_name='required_sample_info'
+                AND table_schema='qiita'
+                AND column_name='{0}'""".format(column))
+
+        if exists_dynamic is not None:
+            conn_handler.execute("""
+                UPDATE qiita.{0}
+                SET {1}={2}
+                WHERE sample_id='{3}'""".format(self._dynamic_table, column,
```

Comment:
Using format here is extremely dangerous as it opens us up to SQL attacks. You can use the string replacement in the execute function to add the value and id.